gen3/neural-networks/advanced-examples/ocr/ocr/utils/host_process_detections.py

```diff
@@ -0,0 +1,35 @@
+import depthai as dai
+from depthai_nodes.ml.messages import ImgDetectionExtended
+from typing import List
+
+import numpy as np
+
+class ProcessDetections(dai.node.ThreadedHostNode):
+    def __init__(self):
+        super().__init__()
+        self.detections_input = self.createInput()
+        
+        self.config_output = self.createOutput()
+
+    def run(self) -> None:
+        while self.isRunning():
+            img_detections = self.detections_input.get()
+            detections = img_detections.detections
+            # w, h = img_detections.transformation.getSize()
+            w, h = 1728, 960 
+            
+            configs_message = dai.MessageGroup()
+            for i, detection in enumerate(detections):
+                cfg = dai.ImageManipConfigV2()
+                detection: ImgDetectionExtended = detection
+                rect = detection.rotated_rect
+                rect = rect.denormalize(w, h)
+                cfg.addCropRotatedRect(rect, normalizedCoords=False)
+                cfg.setOutputSize(320, 48)
+                cfg.setReusePreviousImage(False)
+                cfg.setTimestamp(img_detections.getTimestamp())
+                
+                configs_message[str(i+100)] = cfg
```

Comment:
Why the i+100?

Reply:
This is because the message group saves by `string : value` pair and in the ScriptNode when we iterate through this Messagegroup, it is ordered by the string. This just ensures the order is the same. without the `+100` the order of iteratiion of 13 configs (ie 13 detections) would be 0, 1, 10, 11, 12, 2, 3, 4, ... With +100 its 100, 101, 102, 103, ... 112